plugin/evm/config/config.md

```diff
@@ -395,8 +395,12 @@ Specifies the maximum amount of CPU time that can be stored for a single WS conn
 
 ### `allow-unfinalized-queries`
 
+_Boolean_
+
 Allows queries for unfinalized (not yet accepted) blocks/transactions. Defaults to `false`.
 
+**⚠️ WARNING: Always keep this set to `false` in production.** Setting to `true` causes severe performance degradation as the node must track and serve all unfinalized blocks. Only enable for testing/debugging when specifically needed.
```

Comment:
I would suggest to incorporate the following from Jovica into the warning

"Querying unfinalized blocks makes sense on chains that by design do have reorgs and forks and block confirmations are a necessity so you don't have to wait 10 minutes (or however long) for actual finalization before finding out about a transaction. On Avalanche, block confirmations do not exist, you get a 100% finalized block every 2s, and you do not not need any block to be built on top of a that block to be sure that the block is part of the canonical chain.

So in effect, they are asking you to worsen performance on your end for no actual benefit anywhere. That is not a good approach, and you should push back on that."